common/verify/anchor.py

```
@@ -0,0 +1,167 @@
+            raise RuntimeError(f"eth_getProof HTTP {response.status}")
+        body = await response.json()
+        if not isinstance(body, dict):
+            raise RuntimeError("eth_getProof returned non-object")
+        if "error" in body:
+            raise RuntimeError(f"eth_getProof error: {body['error']}")
+        result = body.get("result")
+        if not isinstance(result, dict):
+            raise RuntimeError("eth_getProof result is not an object")
+        account_proof = result.get("accountProof")
+        if not isinstance(account_proof, list):
+            raise RuntimeError("eth_getProof accountProof is not a list")
```

Ruff (0.15.12)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)